src/DataPersister/LoopDataPersisterInterface.php

```diff
@@ -0,0 +1,25 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\Core\DataPersister;
+
+/**
+ * @author Vincent Chalamon <vincentchalamon@gmail.com>
+ */
+interface LoopDataPersisterInterface
```

Comment:
I’m not a fond of this name, can we found something more explicit such as “StopableDataPersister”, “TransientDataPersister” or “NonFinalDataPersister” or “NonLastDataPersister”?

Also, do we really need this new interface? Can’t we just transform the MessengerDataPersister into a decorator and call the decorated service is persist is set to true? It will simplify the code a lot.

Reply:
The interface helps to continue to the next data provider from the ChainDataProvider, and prevents any direct relation between the ChainDataProvider and the Messenger DataProvider: https://github.com/api-platform/core/pull/3617/files#diff-4c062f54ee87035a3b6a2a1689356789R61

>I’m not a fond of this name, can we found something more explicit such as “StopableDataPersister”, “TransientDataPersister” or “NonFinalDataPersister” or “NonLastDataPersister”?

For code quality and to prevent any useless code complexity, I would prefer something "positive", instead of "stop", "non-...", etc.
I'll try "transient" as you suggested, ok for you too @soyuka @alanpoulain ?